File: POO/Marketplace/Java/Produto.java
class Produto extends Entidade{

     private String nome;
     private float preco;

     public Produto(String nome, float preco){
          this.nome = nome;
          this.preco = preco;
     }

     public String getNome(){
          return this.nome;
     }

     public void setNome(String nome){
          this.nome = nome;
     }

     public float getPreco(){
          return this.preco;
     }

     public void setPreco(float preco){
          this.preco = preco;
     }
}

File: CLP/TP2/produto.rs
#[derive(Clone)]
pub struct Produto{
    pub nome: String,
    pub codigo: u32,
    pub valor: f32,
}

pub trait SetProduto {
    fn set_nome(&mut self, nome: String);
    fn set_codigo(&mut self, codigo: u32);
    fn set_valor(&mut self, valor: f32);
}

pub trait GetProduto {
    fn get_nome(&self) -> String;
    fn get_codigo(&self) -> u32;
    fn get_valor(&self) -> f32;
}

impl GetProduto for Produto {
    fn get_nome(&self) -> String{
        return self.nome.clone();
    }

    fn get_codigo(&self) -> u32{
        return self.codigo;
    }

    fn get_valor(&self) -> f32{
        return self.valor;
    }
}

impl SetProduto for Produto {
    fn set_nome(&mut self, nome: String){
        self.nome = nome;
    }

    fn set_codigo(&mut self, codigo: u32){
        self.codigo = codigo;
    }

    fn set_valor(&mut self, valor: f32){
        self.valor = valor;
    }
}
